### PR TITLE
Man page  add  guessing parser  subsection

### PR DESCRIPTION
**Pull request by masatake**
This is the first step for the releasing.

### CODE REVIEW

File: man/ctags.1.rst.in [hunk 1]
```diff
@@ -20,54 +20,174 @@ DESCRIPTION
...
+	rule of "interpreter" testing if the lines at the tail of the file
+	have the following pattern::
+
+		Local Variables:
+		..
+		mode: MODE
+		...
+		End:
+
+	3000 characters are seeked from the end of file to find the pattern.
```
Comment (k-takata) [inline man/ctags.1.rst.in]: seeked -> sought

File: man/ctags.1.rst.in [hunk 2]
```diff
@@ -20,54 +20,174 @@ DESCRIPTION
...
+either a single directory (e.g.  "@CTAGS_NAME_EXECUTABLE@ \*"), or on
+all files in an entire source directory tree
+(e.g. "@CTAGS_NAME_EXECUTABLE@ -R"), since only those files whose
 names are mapped to languages will be scanned.
 
-[The reason that .h extensions are mapped to C++ files rather than C files
-is because it is common to use .h extensions in C++, and no harm
-results in treating them as C++ files.]
+The same extensions are mapped to multiple parsers. For example, ".h"
+are mapped to C++, C and ObjectiveC. These mappings can cause a
```
Comment (masatake) [inline man/ctags.1.rst.in]: "ObjectiveC" is the name of parser in ctags for parsing input files written in Objective-C language.
Plesae, see `--list-languages` output.

Comment (k-takata) [inline man/ctags.1.rst.in]: Ah, okay.

File: configure.ac [hunk 3]
```diff
@@ -703,6 +703,8 @@ AC_CHECK_FUNCS([mblen], [], [HAVE_MBLEN=1], [])
 
 AC_CONFIG_FILES([Makefile
 		 man/ctags.1.rst
+		 man/ctags-incompatibilities.1.rst
+		 man/ctags-optlib.1.rst
```
Comment (k-takata) [inline configure.ac]: Should ctags-incompatibilities and ctags-optlib be in the section 1?
Isn't it better to put in the section 7?
https://en.wikipedia.org/wiki/Man_page#Manual_sections

It's also better to mention these pages in the "SEE ALSO" subsection.

File: man/ctags.1.rst.in [hunk 4]
```diff
@@ -20,54 +20,174 @@ DESCRIPTION
...
+@CTAGS_NAME_EXECUTABLE@ is capable of generating different kinds of tags
 for each of many different languages. For a complete list of supported
 languages, the names by which they are recognized, and the kinds of tags
 which are generated for each, see the ``--list-languages`` and ``--list-kinds``
 options.
 
+This man page describes Universal-ctags, an implementation of ctags
+derived from Exuberant-ctags. The major incompatible changes between
+Universal-ctags and Exuberant-ctags are enumerated in
+ctags-incompatibilities(1).
```
Comment (k-takata) [inline man/ctags.1.rst.in]: If you change the section number, this should be updated.

File: man/ctags.1.rst.in [hunk 5]
```diff
@@ -20,54 +20,174 @@ DESCRIPTION
...
+
+One of the advantages of Exuberant-ctags is that it allows an user to
+define a new parser from command line. Extending this feature is one
+of the major topics in Universal-ctags development. ctags-optlib(1)
```
Comment (k-takata) [inline man/ctags.1.rst.in]: As above.

File: man/ctags.1.rst.in [hunk 6]
```diff
@@ -20,54 +20,174 @@ DESCRIPTION
...
 platforms which support it, if the name of a file is not mapped to a
-language and the file is executable, the first line of the file is checked
-to see if the file is a "#!" script for a recognized language.
-
-By default, all other files names are ignored. This permits running
-@CTAGS_NAME_EXECUTABLE@ on all files in either a single directory (e.g.
-"@CTAGS_NAME_EXECUTABLE@ \*"), or on all files in an entire source directory
-tree (e.g. "@CTAGS_NAME_EXECUTABLE@ -R"), since only those files whose
+language, @CTAGS_NAME_EXECUTABLE@ tries to guess the language for
+the file by inspecting its content. See "GUESSING PARSER".
```
Comment (k-takata) [inline man/ctags.1.rst.in]: "GUESSING PARSER" -> "Guessing parser"
Cases are not matched with the following subsection name.

**Comment by k-takata** [comment]
The last commit message starts with "fixup!". Maybe you intended to squash it with the previous commit?

**Comment by masatake** [comment]
> The last commit message starts with "fixup!". Maybe you intended to squash it with the previous commit?
Yes, I will do "git rebase -i --autosquash master".
 I must remember  doing s/1/7/ in the commit log:-).

**Comment by masatake** [comment]
Thank you very much.  I will merge this changes.

**Pull request merged**